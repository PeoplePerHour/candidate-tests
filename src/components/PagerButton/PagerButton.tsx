import * as React from "react";
import {PropsWithChildren} from 'react'


interface IProps{
    handleNavigate(page:string,num:number):void
    text:string
    querySting?:any
    link?:any
}

const PagerButton:React.FC<IProps> = (props:PropsWithChildren<IProps>) =>{

const {link,handleNavigate,text}=props

const handleClick=(e: React.MouseEvent<HTMLButtonElement, MouseEvent>)=>{
    e.preventDefault()
    let url = new URL(link);
    let params = new URLSearchParams(url.search.slice(1));
    let num = parseInt(params.get("page"));
    handleNavigate('page',num)
}
return <button className='btn' disabled={link==""?true:false} onClick={(e)=>handleClick(e)}>{text}</button>
};

export default PagerButton;
