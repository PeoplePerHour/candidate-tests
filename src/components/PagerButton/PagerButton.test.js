
import * as  React from 'react';
import * as ReactDOM from 'react-dom';

import PagerButton from './PagerButton'


it('renders PagerButton without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<PagerButton/>, div);
    expect(div.firstChild).toMatchSnapshot();
    ReactDOM.unmountComponentAtNode(div);
})
