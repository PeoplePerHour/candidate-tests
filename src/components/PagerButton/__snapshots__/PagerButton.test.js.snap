// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PagerButton without crashing 1`] = `
<button
  class="btn"
/>
`;
